xdsl/dialects/qref.py

```diff
@@ -30,13 +31,36 @@ class QubitAttr(ParametrizedAttribute, TypeAttribute):
 
 
 class QRefBase(IRDLOperation, ABC):
-    pass
+    @abstractmethod
+    def ssa_op(self) -> qssa.QssaBase:
+        """
+        Get corresponding qssa operation
+        """
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def is_gate(self) -> bool:
+        """
+        Is this operation a gate
+        """
+        raise NotImplementedError()
 
 
 @irdl_op_definition
 class QRefAllocOp(QRefBase):
     name = "qref.alloc"
 
+    def ssa_op(self) -> qssa.QubitAllocOp:
+        return qssa.QubitAllocOp.create(
+            result_types=[qssa.qubit] * self.num_qubits,
+            attributes=self.attributes,
+        )
+
+    @property
+    def is_gate(self) -> bool:
+        return False
+
     res: VarOpResult = var_result_def(qubit)
```

Comment:
Style comment: Can you move `res` before the methods please

Reply:
@AntonLydike I maybe incorrectly believed that this would cause some style check to fail, since it is done this way across all xdsl ops I've seen.